dev.c

```diff
@@ -135,20 +134,70 @@ static unsigned len_args(unsigned numargs, struct fuse_arg *args)
 
 static u64 fuse_get_unique(struct fuse_conn *fc)
 {
-	fc->reqctr++;
+	struct fuse_per_cpu_ids *my_ids;
+	u64 uid;
+	int num_alloc;
+
+	int cpu = get_cpu();
+
+	my_ids = per_cpu_ptr(fc->per_cpu_ids, cpu);
+
+	if (unlikely(my_ids->num_free_ids == 0)) {
+		spin_lock(&fc->lock);
+		BUG_ON(fc->num_free_ids == 0);
```

Comment:
What guarantees fc->num_free_ids != 0? Why wouldn't u run out of ids if u have a large number of cpus and a subset of them have consumed most of the free_ids. And fuse_get_unique() get called on a cpu which doesn't have free_ids stack yet?

Reply:
We allocate twice the number of ids as the maximum number of pending requests to allow for per cpu allocations. This is good for 512 CPUs.